Source/Resources/MongoDB/DolittleMongoConventions.cs

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) Dolittle. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using MongoDB.Bson.Serialization.Conventions;
+
+namespace Dolittle.SDK.Resources.MongoDB;
+
+/// <summary>
+/// The Dolittle Mongo Conventions.
+/// </summary>
+public static class DolittleMongoConventions
+{
+    static readonly object _registerLock = new();
+    static bool _isRegistered;
+    
+    /// <summary>
+    /// Ensured that the default Mongo Conventions are registered;
+    /// </summary>
+    public static void EnsureConventionsAreRegistered()
+    {
+        if (_isRegistered)
+        {
+            return;
+        }
```

Comment:
This operation shouldn't really take that long. So I think this early checking isn't all that necessary here.
```suggestion
```

Reply:
I would maybe prefer to have it in, so that it follows the usual pattern of check => lock => check